cookbook/persistance/key-value.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: page
+title: "Storing key-value data on disk"
+permalink: /cookbook/persistence/key-value/
+---
+
+If we have a relatively small collection of key-values that we'd like to save, we can use the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin.
+
+Normally we would have to write native platform integration for storing data for both platforms. Fortunately the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin can be used for that. Shared preferences plugin wraps `NSUserDefaults` on iOS and `SharedPreferences` on Android, providing a persistent store for simple data.
+
+## Setup
+
+Before we start, we need to add [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin to our `pubspec.yaml` file:
+
+```yaml
+dependencies:
+  flutter:
+    sdk: flutter
+  shared_preferences: "<newest version>"
+```
+
+## Saving data
+
+In order to access store, we can use `SharedPreferences` class.
+
+
+```dart
+// import shared preferences plugin
+import 'package:shared_preferences/shared_preferences.dart';
+
+// obtain shared preferences 
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+// set new value
+prefs.setInt('counter', counter);
+```
+
+
+## Reading data
+
+```dart
+import 'package:shared_preferences/shared_preferences.dart';
+
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+int counter = (prefs.getInt('counter') ?? 0) + 1;
+```
+
+## Removing data
+
+```dart
+import 'package:shared_preferences/shared_preferences.dart';
+
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+prefs.remove('counter');
+```
+
+In example above we load data from `counter` key, if it not exists, `0` is returned. 
+
+For saving data we call `set` method. Note that data is persisted asynchronously. If we want to get notified when data is saved use `commit()` function. 
```

Comment:
Does this content belong in the "Saving Data" section? I think you could combine it with the "To persist key-value data, we can use..." line